packages/modal-ui/src/lib/components/DerivationPath.tsx

```diff
@@ -229,71 +205,100 @@ export const DerivationPath: React.FC<DerivationPathProps> = ({
 
   return (
     <div className="derivation-path-wrapper">
-      {showMultipleAccountsSelect ? (
+      {route === "EnterDerivationPath" && (
+        <div className="enter-derivation-path">
+          <div>
+            <p>
+              Make sure your device is plugged in, then enter a derivation path
+              to connect:
+            </p>
+            <input
+              type="text"
+              placeholder="Derivation Path"
+              value={derivationPath}
+              onChange={(e) => {
+                setDerivationPath(e.target.value);
+              }}
+              onKeyPress={handleEnterClick}
+            />
+          </div>
+          <div className="action-buttons">
+            <button className="left-button" onClick={onBack}>
+              Back
+            </button>
+            <button className="right-button" onClick={handleValidateAccount}>
+              Continue
+            </button>
+          </div>
+        </div>
+      )}
+
+      {route === "ChooseAccount" && (
         <HardwareWalletAccountsForm
-          hardwareWalletAccounts={hardwareWalletAccounts}
-          onAccountChanged={(derivationPath, selectedAccountId) => {
-            handleAccountChange(derivationPath, selectedAccountId);
+          accounts={accounts}
+          onSelectedChanged={(index, selected) => {
+            setAccounts((prevAccounts) => {
+              const updateAccounts = prevAccounts.map((account, idx) => {
+                const selectedValue =
+                  index === idx ? selected : account.selected;
+                return {
+                  ...account,
+                  selected: selectedValue,
+                };
+              });
+              return [...updateAccounts];
+            });
           }}
-          onSubmit={(accounts, e) => {
+          onSubmit={(acc, e) => {
             e.preventDefault();
-            const mapAccounts = accounts.map((account) => {
-              return {
-                derivationPath: account.derivationPath,
-                publicKey: account.publicKey,
-                accountId: account.selectedAccountId,
-              };
+            setAccounts((prevAccounts) => {
+              prevAccounts = prevAccounts.filter((account) => account.selected);
+
+              return [...prevAccounts];
```

Comment:
We should avoid reassigning here and simply spread the filter or give it a name like `selectedAccounts`